source/m2mtlvserializer.cpp

```diff
@@ -164,9 +177,44 @@ bool M2MTLVSerializer::serialize_multiple_resource(const M2MResource *resource,
     return success;
 }
 
+/* See, OMA-TS-LightweightM2M-V1_0-20170208-A, Appendix C,
+ * Data Types, Integer, TLV Format 
+ * 
+ * This is easy to change to support 8, 16 or 32 interger. */
+template <typename T> void M2MTLVSerializer::set_value_int(T value, uint8_t *buffer, uint32_t &size)
+{
+    size = sizeof(value);
+
+    /* write bytes to big endian order in buffer */
+    for (uint8_t ind=0;ind<size;ind++) {
+        buffer[size-ind-1] = ((value >> (ind*8)) & 0xFF);
+    }
+}
+
+template void M2MTLVSerializer::set_value_int<int8_t>(int8_t value, uint8_t *buffer, uint32_t &size);
```

Comment:
I think correct interface functions for 16 and 32 bits in this case are htons and htonl http://man.yolinux.com/cgi-bin/man2html?cgi_command=htons&cgi_section=3. These seems to be available under mbed-os. For Linux there is also available common 64-bit version http://man.yolinux.com/cgi-bin/man2html?cgi_command=htobe64&cgi_section=.  

Maybe PAL should offer these common functions by above for all targets.